rmw_connext_shared_cpp/src/node_names.cpp

```diff
@@ -35,19 +36,17 @@ get_node_names_impl(
   rcutils_string_array_t * node_namespaces,
   rcutils_string_array_t * enclaves)
 {
-  if (!node) {
-    RMW_SET_ERROR_MSG("node handle is null");
-    return RMW_RET_ERROR;
-  }
-  if (node->implementation_identifier != implementation_identifier) {
-    RMW_SET_ERROR_MSG("node handle is not from this rmw implementation");
-    return RMW_RET_ERROR;
+  RMW_CHECK_ARGUMENT_FOR_NULL(node, RMW_RET_INVALID_ARGUMENT);
+  RMW_CHECK_TYPE_IDENTIFIERS_MATCH(
+    node,
+    node->implementation_identifier,
+    implementation_identifier,
+    return RMW_RET_INCORRECT_RMW_IMPLEMENTATION);
+  if (RMW_RET_OK != rmw_check_zero_rmw_string_array(node_names)) {
+    return RMW_RET_INVALID_ARGUMENT;
   }
```

Comment:
this seems inconsistent with:
```cpp
  ret = rmw_names_and_types_check_zero(topic_names_and_types);
  if (RMW_RET_OK != ret) {
    return ret;
  }
```

Reply:
My guess is that `rmw_check_zero_rmw_string_array(node_names)` does not return `RMW_RET_INVALID_ARGUMENT`

Reply:
ok, it's a bit weird the semantic different between the two, but it doesn't matter much,

Reply:
Indeed. `rmw_check_zero_rmw_string_array` returns `RMW_RET_INVALID_ARGUMENT` if the argument is NULL, and `RMW_RET_ERROR` if it's not zero initialized. We can change `rmw_names_and_types_check_zero` call sites to the do the same for consistency's sake though.